backend/director/agents/web_search_agent.py

```
@@ -0,0 +1,183 @@
+            )
+            self.output_message.content.append(videos_content)
+            self.output_message.push_update()
+
+            logger.info(f"Found {len(valid_videos)} valid videos.")
+            return AgentResponse(
+                status=AgentStatus.SUCCESS,
+                message="Video search completed successfully.",
+                data={"videos": [video.dict() for video in valid_videos]},
+            )
+        except requests.exceptions.RequestException as e:
+            error_code = getattr(e.response, 'status_code', None) if hasattr(e, 'response') else None
```

Ruff (0.8.2)

Local variable `error_code` is assigned to but never used

Remove assignment to unused variable `error_code`

(F841)